### DISCUSSION TITLE
How to directly connect in wallet app?

### DISCUSSION THREAD

**Discussion by mfer725** [rainbow-me/rainbowkit#1314] [Q&A] [answered]
If it is in the built-in browser of a wallet app (such as MetaMask, TrustWallet, TokenPocket ...), I hope to be able to directly connect to the wallet instead of a pop-up modal. How can this be achieved?

**Comment by glitch-txs** [comment]
You can do this with wagmi, rainbowkit is the modal built on top of wagmi

**Comment by DanielSinclair** [comment, marked as answer]
Wallets like MetaMask require that a user approves a "connection" before exposing a wallet (aka account) to the dApp, so a user will always need to interact with the wallet's popup. Some browsers like Safe don't require this and are a bit cleaner to use. Once a user approves the connection, when they visit again in the in-app browser or with a browser extension, their approval is typically remembered and they will auto-connect as long as Wagmi is configured with:
```ts
const config = createConfig({
  autoConnect: true,
})
```
